opennms-dao/src/test/java/org/opennms/netmgt/dao/support/JdbcFilterDaoIT.java

```diff
@@ -353,4 +362,36 @@ public void testVariousWaysToMatchServiceNames() {
         // Just make sure this one doesn't hurl
         m_dao.getInterfaceWithServiceStatement("serviceName == 'DiskUsage-/foo/bar'");
     }
+
+    // Verifies that if a bunch of interfaces exists, checking if an ip address is valid should be faster
+    // than retrieving all interfaces.
+    // See HZN-1161 for more details.
+    @Test
+    public void verifyPerformance() {
+        // Create a bunch of interfaces
+        final OnmsNode node1 = m_populator.getNode1();
+        final IPAddressRange ipAddresses = new IPAddressRange("10.10.0.0", "10.10.255.255");
+        final Iterator<IPAddress> iterator = ipAddresses.iterator();
+        while (iterator.hasNext()) {
+            IPAddress address = iterator.next();
+            OnmsIpInterface ipInterface = new OnmsIpInterface();
+            ipInterface.setNode(node1);
+            ipInterface.setIpAddress(address.toInetAddress());
+            m_interfaceDao.save(ipInterface);
+        }
+        assertThat(m_interfaceDao.countAll(), greaterThan(255 * 255));
+
+        // measure
+        long start = System.currentTimeMillis();
+        m_dao.getActiveIPAddressList("IPADDR != '0.0.0.0'");
+        long diff = System.currentTimeMillis() - start;
+
+        // measure
+        start = System.currentTimeMillis();
+        assertThat(m_dao.isValid("10.10.0.1", "IPADDR != '0.0.0.0'"), is(true));
+        long diff2 = System.currentTimeMillis() - start;
+
+        // verify
+        assertThat(diff, greaterThan(diff2));
```

Comment:
Comparing timing in this fashion may cause flaps i.e. if the JVM GC pauses during the 2nd measure, then diff2 might be > diff.

I would just print out the different time stamps and include assertion for the functional bits.

Reply:
Then we can get rid of the test as it does not make sense to have it (: